app/controllers/course/statistics/aggregate_controller.rb

```diff
@@ -2,9 +2,11 @@
 # This is named aggregate controller as naming this as course controller leads to name conflict issues
 class Course::Statistics::AggregateController < Course::Statistics::Controller
   before_action :preload_levels, only: [:all_students, :course_performance]
+  before_action :fetch_published_assessments, only: [:all_assessments, :submission_due]
```

Comment:
```suggestion
  before_action :preload_assessments, only: [:all_assessments, :submission_due]
```
be consistent in the naming. see line 4 above.

Reply:
got it.. though I will change it into `preload_published_assessments` since we actually only care about the published one, and ignoring the unpublished.